packages/client/src/runtime/core/engines/library/WasmLibraryLoader.ts

```diff
@@ -31,20 +25,25 @@ export const wasmLibraryLoader: LibraryLoader = {
     // bindgen keeps an internal cache of its instance already, when the wasm
     // engine is loaded more than once it crashes with `unwrap_throw failed`.
     if (loadedWasmInstance === undefined) {
-      const wasmMod = await config.getQueryEngineWasmModule?.()
-
-      if (wasmMod === undefined || wasmMod === null) {
-        throw new PrismaClientInitializationError(
-          'The loaded wasm module was unexpectedly `undefined` or `null` once loaded',
-          clientVersion,
-        )
-      }
-
-      // from https://developers.cloudflare.com/workers/runtime-apis/webassembly/rust/#javascript-plumbing-wasm-bindgen
-      loadedWasmInstance = new WebAssembly.Instance(wasmMod, { './query_engine_bg.js': wasmBindgenRuntime }).exports
-      wasmBindgenRuntime.__wbg_set_wasm(loadedWasmInstance)
+      loadedWasmInstance = (async () => {
+        const wasmModule = await config.getQueryEngineWasmModule?.()
+
+        if (wasmModule === undefined || wasmModule === null) {
+          throw new PrismaClientInitializationError(
+            'The loaded wasm module was unexpectedly `undefined` or `null` once loaded',
```

Comment:
When could this case happen?

Reply:
Practically, I don't think it could ever happen unless a bundler did something awful to the bundle, and even then it's likely to be a different error, but here because it's technically possible according to the type system.